### DISCUSSION TITLE
Choice selections not being sent in on form submit, radio works

### DISCUSSION THREAD

**Discussion by aseem-md** [bhch/django-jsonform#146] [Q&A]
Hi All, 

I have been trying to figure out what is happening here and have not quiet been able to figure out what is going on? I have a JSOField in a model, have a schema, in which there is a field for gender which is specified as: 

`"gender": {
                "type": "string",
                "choices": ["M", "F", "O"],
            },
`

What I noticed was that when the form gets saved, 

- any changes I make to this field just gets ignored and the old value is being submitted. 
- An array of items each of which is a choice it works fine and the updated list is submitted.
- If I use the radio widget the new value is submitted. (This is only tenable for small lists, I would like to use choice without radio buttons for larger choices list)

Perhaps I am missing something in the schema definition that would allow this to work? 

Any help would be appreciated.

Thanks.

**Comment by bhch** [comment]
Could you give me a sample schema to reproduce it?

**Comment by aseem-md** [comment]
I will try to get a repo to you eod. Also I should mention that this is inside django-admin if that makes any difference. And if I cannot reproduce on a smaller scale, I will have to go spelunking. Thanks.

**Reply by aseem-md** [reply]
Apparently the minimum reduction case for django-admin works :/ Might have to debug form submission.

**Comment by aseem-md** [comment]
Hi,

So I created a reproduction case and was not able to reproduce data from the form not being submitted properly. 

However this still seems to happen on the main app that I have. The differences I noticed were that in the [repro case](https://github.com/aseem-md/django-jsonform-test)  that the fields submitted were the `rjf..` ones and a correct, updated jsonfield value.

The initial smaller image attached shows this. 
![2024-02-09-110833_176x168_scrot](https://github.com/bhch/django-jsonform/assets/116107918/1a98873e-eea1-47f9-9de8-3971c95073fa)

However when I use the same code structure in my project, I get the `rjf...` variables that have the correct updated values, the `builder` variable that _does not_ have the correct updated value and there is an additional field called `initial-builder` which does not show up in the repro case. 
![2024-02-09-110949_78x777_scrot](https://github.com/bhch/django-jsonform/assets/116107918/9e3ec77d-9d58-4359-851d-cecfcbabfd27)


Perhaps I am missing something very basic related to how forms get processed?

Any help would be greatly appreciated.

**Reply by bhch** [reply]
You can ignore the `rjf...` variables. These are essentially names of the input fields generated by the widget in the browser. It is a long standing bug (https://github.com/bhch/react-json-form/issues/6) but these are not supposed to be submitted. I'm going to conclusively fix it in the next release.

Only variable of concern is the name of json model field. That will contain the actual json data for the field.

So can I try the repo you've linked to reproduce it?

**Reply by aseem-md** [reply]
Yes, that was specifically for debugging this. `poetry` based so you should be able to build and run it as is. 

However I have not been able to reproduce the issue I am seeing in my code base although they are both doing exactly the same thing. For whatever reason the field named `builder` (in the second screenshot) does not contain the updated data, just the old data. In the attempted repro, modifying the the form, does submit the updated data in the `bld` field.

The attempted repro case also does not contain the initial-data field which is concerning, but it seems to work. Unfortunately I don't have a path forward to debug this as yet. 

Thanks.

**Reply by aseem-md** [reply]
Happy to share my json schema via alternative means if that helps.

**Reply by aseem-md** [reply]
FYI On further investigation, in my schema, choices that are part of an array seem to work fine, however standalone choices do not get submitted to the POST.

**Reply by aseem-md** [reply]
Another FYI for the string enum drop downs if I convert them to an array type it seems to submit the right thing and I can transform it back and forth. I am not sure how I can reproduce it, but I will try and keep you posted.

**Reply by bhch** [reply]
I tried the sample repo. I'm unable to reproduce the behaviour and things are working as expected.

> Happy to share my json schema via alternative means if that helps.

If it can help reproducing the bug, you can mail it to me. My email is mentioned on my profile page.
